src/wallet/api/wallet.cpp

```diff
@@ -2095,21 +2095,21 @@ bool WalletImpl::verifyMessageWithPublicKey(const std::string &message, const st
 
 bool WalletImpl::connectToDaemon()
 {
-    bool result = m_wallet->check_connection(NULL, NULL, DEFAULT_CONNECTION_TIMEOUT_MILLIS);
-    if (!result) {
-        setStatusError("Error connecting to daemon at " + m_wallet->get_daemon_address());
+    const boost::system::error_code err = m_wallet->check_connection(NULL, NULL, DEFAULT_CONNECTION_TIMEOUT_MILLIS);
+    if (err) {
+        setStatusError("Error connecting to daemon at " + m_wallet->get_daemon_address() + " : " + err.message());
     } else {
         clearStatus();
         // start refreshing here
     }
-    return result;
+    return !err;
 }
 
 Wallet::ConnectionStatus WalletImpl::connected() const
 {
     uint32_t version = 0;
     bool wallet_is_outdated = false, daemon_is_outdated = false;
-    m_is_connected = m_wallet->check_connection(&version, NULL, DEFAULT_CONNECTION_TIMEOUT_MILLIS, &wallet_is_outdated, &daemon_is_outdated);
+    m_is_connected = bool(m_wallet->check_connection(&version, NULL, DEFAULT_CONNECTION_TIMEOUT_MILLIS, &wallet_is_outdated, &daemon_is_outdated));
```

Comment:
Shouldn't this be `!bool()` since error code 0 means success?

Reply:
Good catch. Initially this returned `expect<void>`, but decided to keep things consistent with `error_code`.